contracts/src/test/7683/T1XChainReadTest.t.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import { TypeCasts } from "@hyperlane-xyz/libs/TypeCasts.sol";
+import { ITransparentUpgradeableProxy } from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import { OrderData, OrderEncoder } from "intents-framework/libs/OrderEncoder.sol";
+import { OnchainCrossChainOrder } from "intents-framework/ERC7683/IERC7683.sol";
+
+import { T1XChainReader } from "../../libraries/xChain/T1XChainReader.sol";
+import { T1XChainMessage } from "../../libraries/xChain/T1XChainMessage.sol";
+import { T1BasicSwapE2E } from "./T1BasicSwapE2E.t.sol";
+import { T1ERC7683Pull } from "../../7683/T1ERC7683Pull.sol";
+
+contract T1XChainReaderTest is T1BasicSwapE2E {
+    using TypeCasts for address;
+
+    T1XChainReader internal originReader;
+    T1XChainReader internal destinationReader;
+    T1ERC7683Pull internal L1T17683Pull;
+    T1ERC7683Pull internal L2T17683Pull;
+
+    function setUp() public virtual override {
+        super.setUp();
+
+        // Deploy T1XChainReader on both chains
+        originReader = T1XChainReader(payable(_deployProxy(address(0))));
+        admin.upgrade(
+            ITransparentUpgradeableProxy(address(originReader)),
+            address(new T1XChainReader(address(l1t1Messenger), origin))
+        );
+
+        destinationReader = T1XChainReader(payable(_deployProxy(address(0))));
+        admin.upgrade(
+            ITransparentUpgradeableProxy(address(destinationReader)),
+            address(new T1XChainReader(address(l2t1Messenger), destination))
+        );
+
+        L1T17683Pull = T1ERC7683Pull(payable(_deployProxy(address(0))));
+        L2T17683Pull = T1ERC7683Pull(payable(_deployProxy(address(0))));
+        admin.upgrade(
+            ITransparentUpgradeableProxy(address(L1T17683Pull)),
+            address(new T1ERC7683Pull(address(l1t1Messenger), address(0), address(originReader), uint32(origin)))
+        );
+        admin.upgrade(
+            ITransparentUpgradeableProxy(address(L2T17683Pull)),
+            address(
+                new T1ERC7683Pull(address(l2t1Messenger), address(0), address(destinationReader), uint32(destination))
+            )
+        );
+        L1T17683Pull.initialize(address(L2T17683Pull));
+        L2T17683Pull.initialize(address(L1T17683Pull));
+    }
+
+    // 1. user opens intent on source chain
+    // 2. solver fills intent on destination chain
+    // 3a. solver calls 7683 verifySettlement on source chain, triggering T1XChainReader.requestRead
+    // 4a. relayer picks up message and calls getFilledOrderStatus on destination chain
+    // 4b. relayer calls T1XChainReader.handle with the result of the read which calls
+    // onT1XChainReaderResult on callback address
+    // 4c. onT1XChainReaderResult on 7683 contract settles intent and releases funds to solver
+    function test_ERC7683PullSettlementFlow() public {
```

Comment:
A high level question, is 𝚝𝟷 network ever visited in this flow?

Using L2 prefix is kind of confusing when having a more than one (𝚝𝟷 & PR1). Wdyt? Shall we revisit the naming in a different PR?

Reply:
it's not really important what either chain is in this test